### PR TITLE
Clean up directory reference

### PR DESCRIPTION
**Pull request by madhurajayaraman**
b/302730696

### CODE REVIEW

File: starboard/shared/win32/posix_emu/dirent.cc [hunk 1]
```diff
@@ -180,18 +184,25 @@ DIR* opendir(const char* path) {
 }
 
 int closedir(DIR* dir) {
+  if (!dir)
+    errno = EBADF;
+  return -1;
```
Comment (kaidokert) [inline starboard/shared/win32/posix_emu/dirent.cc:189]: This looks like you maybe missed the braces around `if(!dir)` here. I don't think there should be an unconditional `return -1`

Comment (madhurajayaraman) [reply starboard/shared/win32/posix_emu/dirent.cc:189]: Gah, good catch! Ty

File: base/files/file_enumerator_unittest.cc [hunk 2]
```diff
@@ -419,12 +419,15 @@ TEST(FileEnumerator, InvalidDirectory) {
...
 
-  // Slightly different outcomes between Windows and POSIX.
-#if BUILDFLAG(IS_WIN)
-  EXPECT_EQ(File::Error::FILE_ERROR_FAILED, enumerator.GetError());
-#else
-  EXPECT_EQ(File::Error::FILE_ERROR_NOT_A_DIRECTORY, enumerator.GetError());
-#endif
+// Different behaviour between directory APIs and posix mappings
+// specifically in how we set error codes
+//
```
Comment (kaidokert) [inline base/files/file_enumerator_unittest.cc:424]: Did you want to leave this commented out here ? If this isn't resolvable in this PR, maybe lets create a small follow-up bug to clean this up and link here as a TODO(b/1234)

File: base/files/file_enumerator_starboard.cc [hunk 3]
```diff
@@ -112,10 +112,8 @@ FileEnumerator::~FileEnumerator() = default;
 std::vector<FileEnumerator::FileInfo> FileEnumerator::ReadDirectory(
     const FilePath& source) {
   internal::AssertBlockingAllowed();
-  SbFileError error;
-  SbDirectory dir = SbDirectoryOpen(source.value().c_str(), &error);
-  if (!SbDirectoryIsValid(dir)) {
-    error_ = static_cast<File::Error>(error);
+  DIR* dir = opendir(source.value().c_str());
+  if (!dir) {
     return std::vector<FileEnumerator::FileInfo>();
```
[the remark this answers is not in the document]
Comment (madhurajayaraman) [reply base/files/file_enumerator_starboard.cc:117]: The error we cast there is the SbFileError that we used to set in SbDirectoryOpen. We need to re-handle all of the error codes so I can make a note of this as a place to re-implement something but I'm not sure what the appropriate handling should be instead.

File: base/files/file_enumerator_starboard.cc [hunk 4]
```diff
@@ -133,15 +131,25 @@ std::vector<FileEnumerator::FileInfo> FileEnumerator::ReadDirectory(
...
   std::vector<FileEnumerator::FileInfo> ret;
-  // We test if SbDirectoryGetNext returns the parent directory, i.e. |..|,
+  // We test if readdir_r returns the parent directory, i.e. |..|,
   // because whether or not it is returned is platform-dependent and we need to
   // be able to guarantee it is returned when the INCLUDE_DOT_DOT bitflag is
   // set.
   bool found_dot_dot = false;
 
   std::vector<char> entry(kSbFileMaxName);
+  struct dirent dirent_buffer;
```
Comment (kaidokert) [inline base/files/file_enumerator_starboard.cc:141]: No need for action or changes in this PR, but just leaving a comment.

We should probably aim to entirely `_delete_` `file_enumerator_starboard.cc` and simply use `file_enumerator_posix.cc` wholesale, without any Starboard #ifdefs . To that end, i would reference how the loop is written in that module and probably track it a little closer here.

But - it's fine to get this in as is, if tests are passing. Just a call-out that we seem to have substantially different behavior still.

File: base/files/file_enumerator_starboard.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (madhurajayaraman) [reply base/files/file_enumerator_starboard.cc:118]: Ahh gotcha. Modified.

**Review by y4vor — APPROVED** [review]

File: starboard/directory.h [hunk 5]
```diff
@@ -42,8 +42,8 @@ static inline bool SbDirectoryIsValid(SbDirectory directory) {
   return directory != kSbDirectoryInvalid;
 }
 
-// DEPRECATED with SB_API_VERSION 16
-//
+#if SB_API_VERSION < 16
```
Comment (kaidokert) [inline starboard/directory.h:45]: This should now be changed to `SB_API_VERSION < 17`

**Pull request closed without merging**